### DISCUSSION TITLE
Questions for CONNECTION in COPY and CREATE STAGE

### DISCUSSION THREAD

**Discussion by Xuanwo** [databendlabs/databend#6620] [Q&A] [answered]
I have a few questions about this new feature:

- The existing document already says users can optionally specify an endpoint URL for s3 external located files. See https://databend.rs/doc/reference/sql/dml/dml-copy-into-table#externallocation-for-amazon-s3-like. So now, after being included in the CONNECTION part, is it still optional, or a must now?
- Same question for ACCESS_KEY_ID & SECRET_ACCESS_KEY. The existing document indicates they are optional as well. How about now?

_Originally posted by @soyeric128 in https://github.com/datafuselabs/databend/issues/6573#issuecomment-1183583764_

**Comment by Xuanwo** [comment, marked as answer]
> The existing document already says users can optionally specify an endpoint URL for s3 external located files.

The doc is out-of-date. `ENDPOINT_URL` didn't implement at all.

We will place all connection-related options with the new change into `CONNECTION`.

And `CONNECTION` is optional for now.

> Same question for ACCESS_KEY_ID & SECRET_ACCESS_KEY. The existing document indicates they are optional as well. How about now?

No, `ACCESS_KEY_ID` and `SECRET_ACCESS_KEY` are required now. Databend could return an error if those values are missing. Those values will check based on different services:

- `s3`: optional: `ACCESS_KEY_ID`, `SECRET_ACCESS_KEY`, `ENDPOINT_URL`, `REGION`, `ENABLE_VIRTUAL_HOST_STYLE`
- `azblob`: require `ENDPOINT_URL`, optional `ACCOUT_NAME`, `ACCOUNT_KEY`,

Those restrictions apply to both `COPY` and `CREATE STAGE`.

**Reply by Xuanwo** [reply]
After PR https://github.com/datafuselabs/databend/pull/6623, we support reading from public repo without ak/sk.

**Reply by soyeric128** [reply]
Thank you. Could you review this PR: https://github.com/datafuselabs/databend/pull/6633?
